edgelet/edgelet-docker/src/notary.rs

```diff
@@ -169,33 +168,45 @@ pub fn notary_lookup(
     Error = Error,
 > {
     let mut notary_cmd = Command::new("notary");
+
+    notary_cmd
+        .args(&["lookup", image_gun, tag, "-c"])
+        .arg(config_path);
+
     if let Some(notary_auth) = notary_auth {
         notary_cmd.env("NOTARY_AUTH", notary_auth);
     }
-    notary_cmd
-        .arg("lookup")
-        .args(&[image_gun, tag])
-        .arg("-c")
-        .arg(config_path)
-        .output_async()
-        .then(|notary_output| {
-            let notary_output = notary_output.with_context(|_| {
-                ErrorKind::LaunchNotary("could not spawn notary process".to_owned())
-            })?;
-            let notary_output_string =
-                String::from_utf8(notary_output.stdout).with_context(|_| {
-                    ErrorKind::LaunchNotary("could not retrieve notary output as string".to_owned())
-                })?;
-            debug!("Notary output string is {}", notary_output_string);
-            let split_array: Vec<&str> = notary_output_string.split_whitespace().collect();
-            if split_array.len() < 2 {
-                return Err(ErrorKind::LaunchNotary(
-                    "notary digest split array is empty".to_owned(),
-                )
-                .into());
-            }
-
-            // Notary Server output on lookup is of the format [tag, digest, bytes]
-            Ok((split_array[1].to_string(), lock))
-        })
+
+    let (send, recv) = tokio::sync::oneshot::channel();
+
+    std::thread::spawn(move || {
```

Comment:
What's wrong with tokio-process?

Reply:
It used crossbeam-utils 0.6, which would mean more testing.

Reply:
Ah indeed, via crossbeam-queue. Please edit that info into the PR description before you merge, so that it goes into the commit message.

You could also use [`tokio_threadpool::blocking`](https://docs.rs/tokio-threadpool/0.1.18/tokio_threadpool/fn.blocking.html) since that will use the blocking threadpool that [the runtime already has.](https://docs.rs/tokio/0.1.22/tokio/runtime/struct.Builder.html#method.blocking_threads) But it's not a big deal since we don't expect too many notary processes anyway.

Reply:
I will switch over to `tokio_threadpool::blocking`. I could not find it earlier.